lib/PDF/PDFConfigurator.php

```diff
@@ -0,0 +1,110 @@
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License version 3 as published by the
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\PDF;
+
+use Configurator;
+use InvalidArgumentException;
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+require_once __DIR__ . '/../../modules/Configurator/Configurator.php';
+
+/**
+ * Class PDFConfigurator
+ * @package SuiteCRM\PDF
+ */
+class PDFConfigurator
+{
+    /** @var Configurator */
+    private $configurator;
+
+    /**
+     * SearchConfigurator constructor.
+     *
+     * @param null|Configurator $configurator
+     */
+    public function __construct(Configurator $configurator = null)
+    {
+        if ($configurator === null) {
+            $configurator = new Configurator();
+        }
+
+        $this->configurator = $configurator;
+    }
+
+    /**
+     * Factory method for nice fluent syntax.
+     *
+     * @return PDFConfigurator
+     */
+    public static function make(): PDFConfigurator
+    {
+        return new self();
+    }
+
+    /**
+     * @param string $engine
+     * @return $this
+     */
+    public function setEngine(string $engine): PDFConfigurator
+    {
+        if (empty($engine)) {
+            throw new InvalidArgumentException('$engine cannot be empty');
```

Comment:
the exception will display as
' ' cannot be empty ? 
So, better to replace the variable with a text; say, Engine can not be empty Or, Engine name can not be empty.

Reply:
Agreed, pushed.